xds/src/test/java/io/grpc/xds/internal/rbac/engine/AttributesExtractionTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.rbac.engine;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.envoyproxy.envoy.config.rbac.v2.RBAC;
+import io.grpc.Metadata;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+/** Unit tests for evaluate argument. */
+@RunWith(JUnit4.class)
+public class AttributesExtractionTest {
```

Comment:
Couple of things about this test:

- which class are you testing? EvaluateArgs or AuthorizationEngine ? Ideally only one class. And like I said elsewhere if you had moved extractFields to EvaluateArgs there would be only one class to test

- mocking `EvaluateArgs`'s getter methods is not very useful or trivial. Better would be to mock/spy `ServerCall<?,?> call` member inside `EvaluateArgs` which is more work but is a better test

Reply:
I deleted this test and moved tests for `generateEnvoyAttributes` inside `EvaluateArgsTest` now.